pyfstat/core.py

```diff
@@ -1033,11 +1059,12 @@ def get_fullycoherent_detstat(
                 return self.twoF
             self.get_fullycoherent_log10BSGL()
             return self.log10BSGL
-        self.get_transient_maxTwoFstat(tstart, tend)
-        if self.BSGL is False:
-            return self.maxTwoF
-        else:
-            return self.get_transient_log10BSGL()
+        return self.get_transient_detstats(
+            tstart=tstart,
+            tend=tend,
+            BSGL=getattr(self, "BSGL", False),
+            BtSG=getattr(self, "BtSG", False),
```

Comment:
Can these be `self.BSGL=self.BtSG=False` by default at initialization time?

Reply:
Actually, why give it as inputs if already present as attributes?

Reply:
We're in a bit of a messy situation of sometimes allowing such methods to be called independently of what the setup was (see e.g. the test for this one), sometimes not. So giving the lower-level function these input options, despite its caller in the standard graph using the values from `self`, allows for such more flexible out-of-graph calls. It's probably not great practice, but kinda convenient...

Reply:
Got rid of these as discussed.